src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -963,4 +987,37 @@ bool EventPipe::IsLockOwnedByCurrentThread()
 }
 #endif
 
+// This method will block runtime bring-up IFF DOTNET_DiagnosticsMonitorAddress != nullptr and DOTNET_DiagnosticsMonitorStopOnStart!=0 (it's default state)
+// The s_ResumeRuntimeStartupEvent event will be signaled when the Diagnostics Monitor uses the ResumeRuntime Diagnostics IPC Command
+void EventPipe::PauseForTracingAgent()
+{
+    LIMITED_METHOD_CONTRACT;
+
+    CLRConfigStringHolder pDotnetDiagnosticsMonitorAddress = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DOTNET_DiagnosticsMonitorAddress);
+    if (pDotnetDiagnosticsMonitorAddress != nullptr)
+    {
+        DWORD dwDotnetDiagnosticsMonitorStopOnStart = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_DOTNET_DiagnosticsMonitorStopOnStart);
+        if (dwDotnetDiagnosticsMonitorStopOnStart != 0)
+        {
+            s_ResumeRuntimeStartupEvent = new CLREventStatic();
```

Comment:
If the diagnostic monitor sends the resume command fast enough the startup path will not have reached this line yet. The resume will be ignored and then the process will enter an infinite wait. This event needs to be created before the earliest possible point when the incoming command could signal it.